tests/test_ble_interface_core.py

```
@@ -3332,6 +3362,51 @@ def test_discard_invalidated_connected_client_clears_pending_when_already_detach
+    """Replacement-pending stale sessions should emit one disconnect publication."""
+    iface = _build_interface(monkeypatch, DummyClient(), start_receive_thread=False)
+    discarded_client = DummyClient()
+    discarded_client.address = "AA:BB:CC:DD:EE:45"
+    discarded_client.bleak_client = SimpleNamespace(address=discarded_client.address)
+    closed_clients: list[BLEClient] = []
+    disconnected_calls: list[bool] = []
+
+    monkeypatch.setattr(
+        iface._client_manager,
+        "_safe_close_client",
+        lambda client: closed_clients.append(client),
```

Ruff (0.15.6)

[warning] Lambda may be unnecessary; consider inlining inner function

Inline function call

(PLW0108)